cache/contenthash/checksum.go

```diff
@@ -570,8 +607,7 @@ func (cc *cacheContext) includedPaths(ctx context.Context, m *mount, p string, o
 			}
 		} else {
 			if !strings.HasPrefix(fn+"/", p+"/") {
-				k, _, kOk = iter.Next()
-				continue
+				break
```

Comment:
could you explain this?

Reply:
At the beginning, we seek to the first path with this prefix (line 541). If we reach a path that don't match the prefix, we can stop iterating, because we're now outside the directory of interest. The old code would keep iterating, but there was no reason to - I think this was left over from before we had that initial seek. So breaking here is an optimization.